DigitalLearningSolutions.Web/Controllers/RegisterController.cs

```diff
@@ -265,7 +253,38 @@ public IActionResult Confirmation()
             return View(viewModel);
         }
 
-        private void ValidateEmailAddress(RegisterViewModel model)
+        private DelegateRegistrationData CreateDelegateRegistrationData(int? centreId)
+        {
+            var delegateRegistrationData = new DelegateRegistrationData
+            {
+                IsCentreSpecificRegistration = centreId.HasValue,
+                Centre = centreId
+            };
+            var id = delegateRegistrationData.Id;
+
+            Response.Cookies.Append(
+                CookieName,
+                id.ToString(),
+                new CookieOptions
+                {
+                    Expires = DateTimeOffset.UtcNow.AddDays(30)
+                }
+            );
+
+            return delegateRegistrationData;
+        }
+
+        private bool CheckCentreIdValid(int? centreId)
+        {
+            if (centreId == null)
+            {
+                return true;
+            }
+
+            return centresDataService.GetCentreName(centreId.Value) != null;

```

Comment:
Could be simplified to `centreId == null || centresDataService.GetCentreName(centreId.Value) != null`;